ReviewChecklist.md

```diff
@@ -0,0 +1,153 @@
+# PR review checklist
+The Solidity compiler is a critical piece of infrastructure in the Ethereum ecosystem and for this
+reason our review process is quite strict.
+The list below is meant to reduce the workload on the core team by helping contributors self-identify
+common issues before they are inevitably pointed out in the review.
+It's also meant as a final checklist for reviewers to go through before approving a PR.
+
+## Before you submit a PR
+- [ ] **Is the issue ready to be worked on?**
+    - If the issue does not have a desirability label (`nice to have`, `should have`,
+        `must have eventually`, `must have`, `roadmap`) we still have not decided whether to implement it.
+    - If the issue `needs design`, we have not yet decided how it should be implemented.
+    - `good first issue candidate` means that the issue will potentially be a `good first issue`
+        eventually but at the moment it's not yet ready to be worked on.
+- [ ] **Is this a breaking change?** Breaking changes should be based on the `breaking` branch rather than on `develop`.
+- [ ] **Does the PR actually address the issue?**
+    - [ ] Mention the issue number in the PR description.
+        If the PR solves it completely, use the `Fixes #<issue number>` form so that Github will close the issue automatically
+    - [ ] Do not include issue number in the PR title, branch name or commit description.
+- [ ] When submitting a PR from a fork **create a branch and give it a good name.**
+    E.g. `fix-array-abi-encoding-bug`.
+    Avoid submitting PRs directly from the `develop` branch of a fork since it makes rebasing and fetching new changes harder.
+- [ ] **Does the PR depend on other PRs?**
+    - [ ] If the PR has dependencies, mention them in bold in the description.
+    - [ ] Avoid basing PRs from forks on branches other than `develop` or `breaking` because
+        unfortunately gitub closes them when the base branch gets merged.
+        Do this only for PRs created directly in the main repo.
+
+## Style and good practices
+- [ ] Does the PR follow our [coding style](CODING_STYLE.md)?
```

Comment:
I mean, Rust got it right with `cargo fmt`, why can't clang? It's not like it's some weird, unusual style.

In any case, we did decide to give `clang-format` another go so hopefully it's just a matter of time. For the time being it's still manual though so we need this point.

Also, the coding style docs also contains some general guidelines that cannot be enforced by the formatter. Ideally we'd be able to point at some popular guide like [C++ Core Guidelines](https://github.com/isocpp/CppCoreGuidelines/blob/master/CppCoreGuidelines.md) but I think that we are a bit more opinionated about some choices :)

Reply:
The main issue is that `clang-format` is incredibly biased toward pre defined styles, e.g. `LLVM`, `Google`, `Microsoft`, which offer a lot of the formatting we want, but don't expose it as separate commands that would allow us to just select a `None` style, and then define whatever we want. The fact that Rust got it right doesn't change the fact that we've implemented the compiler in C++, and are therefore stuck with what we have. Personally, I try to adapt to whatever style the code base is written in, and don't have hard opinions one way or another, but I do think it would be significantly more efficient to just choose the lesser evil, and accept the fact that some things that `clang-format` forces won't be to our liking, as long it actually helps the review process by automating the style checks fully. In any case, this was a bit of a digression, and even more of a rant, so not particularly relevant to this PR.

Reply:
I don't know. To be honest I'm ambivalent about code formatters. Nice to have but I'm used to formatting in a readable way manually and in reviews it's also a pretty insignificant fraction of time spent (even if it's most visible because it accounts for the largest portion of comments). That's why I always opted for readable style, not the most popular one and I'm fully on board with the style we have here. Wake me up when you invent an automatic renamer and commenter so that I can understand the code instantly without taking multiple takes and jumping all over :P